src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,70 +1,251 @@
-                                "torque",
-                                "moabtorque",
-                                "moab",
-                                "pbspro",
-                                "slurm",
-                                "cobalt"
-                            ]
-                        },
-                        "cyclethrottle": {
-                            "type": "integer",
-                            "minimum": 0
-                        },
-                        "taskthrottle": {
-                            "type": "integer",
-                            "minimum": 0
-                        }
-                    },
-                    "required": ["realtime", "scheduler"]
+    "dependency": {
+      "type": "object",
+      "patternProperties": {
+        "^(and|or|not|nand|nor|xor)(_.*)?$": {
+          "$ref": "#/$defs/BoolLoop"
+        },
+        "^datadep(_.*)?$": {
+          "type": "object",
+          "properties": {
+            "attrs": {
+              "type": "object",
+              "properties": {
+                "age": {
+                  "$ref": "#/$defs/TimePattern"
                 },
-                "cycledefs": {
-                    "type": "object",
-                    "properties": {
-                        "groupname": {
-                            "type": "array",
-                            "items": {
-                                "$ref": "#/$defs/dateTimePattern"
-                            }, 
-                            "required": ["ref"]
-                        },
-                        "required": ["items"]
-                    }
+                "minsize": {
+                  "type": "string"
+                }
+              },
+              "additionalProperties": false
+            },
+            "text": {
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        }
+      },
+      "properties": {
+        "taskdep": {
+          "type": "object",
+          "properties": {
+            "task": {
+              "type": "string"
+            },
+            "cycle_offset": {
+              "$ref": "#/$defs/TimePattern"
+            },
+            "state": {
+              "type": "string",
+              "enum": [
+                "RUNNING",
+                "Running",
+                "running",
+                "SUCCEEDED",
+                "DEAD",
+                "Succeeded",
+                "Dead",
+                "succeeded",
+                "dead"
+              ]
+            }
+          },
+          "required": [
+            "task"
+          ]
+        },
+        "streq": {
+          "type": "object",
+          "properties": {
+            "left": {
+              "type": "string"
+            },
+            "right": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "left",
+            "right"
+          ]
```

Comment:
This one should probably have additional properties for consistency.

Reply:
Fair! Inconveniently, as far as I know, `additionalProperties` and `required` have to be specified for every `properties` entry. I'll look into any options for making it defaulted though